dlc-manager/src/manager.rs

```diff
@@ -711,6 +818,43 @@ where
 
         Ok(())
     }
+
+    /// Function to call when we detect that a contract was closed by our counter party.
+    /// This will update the state of the contract and return the [`Contract`] object.
+    pub fn on_counterparty_close(
```

Comment:
I'm thinking it might be better to just watch for a spend of the fund transaction rather than having to call that externally, unless there is a reason you'd prefer it to be external?

Reply:
Sorry I don't get how that's related to BDK. Is the problem that the `Blockchain` interface is sync? But you seem to also use it above to get the current height so isn't that an issue as well?

Reply:
> Is the problem that the Blockchain interface is sync? 

yes

> But you seem to also use it above to get the current height so isn't that an issue as well?

Yeah, bdk stores the most recent sync height so I just use that